packages/core/src/delegation.js

```diff
@@ -121,6 +121,28 @@ export class Delegation {
   get facts() {
     return this.data.facts
   }
+
+  /**
+   * Iterate over the proofs
+   *
+   * @returns {IterableIterator<API.Delegation>}
+   */
+  iterate() {
```

Comment:
I don't see this been used anywhere but a it's own test, which makes me wonder if we really need it. I'm just hesitant to increasing API surface as it's a lot hard to remove things in the future. Unless we really need it I'd much rather leave this out and maybe export static `iterate` below instead. IMO method only useful if different implementations can have different logic in them, here I can't imagine different implementation of `API.Delegation` that would not work with static `iterate`.

Reply:
i use it in w3access, IMO is nicer to have it in the instance we are working with already than to import another method. the static function is still there for any other use or implementation.